sycl/doc/extensions/Reduction/Reduction.md

```diff
@@ -38,6 +38,30 @@ unspecified reduction(span<T, Extent> var, const T& identity, BinaryOperation co
 
 The exact behavior of a reduction is specific to an implementation; the only interface exposed to the user is the set of functions above, which construct an unspecified `reduction` object encapsulating the reduction variable, an optional operator identity and the reduction operator.  For user-defined binary operations, an implementation should issue a compile-time warning if an identity is not specified and this is known to negatively impact performance (e.g. as a result of the implementation choosing a different reduction algorithm).  For standard binary operations (e.g. `std::plus`) on arithmetic types, the implementation must determine the correct identity automatically in order to avoid performance penalties.
 
+Whether an implementation can identify the identity value for a given combination of accumulator type `AccumulatorT` and function object type `BinaryOperation` can be determined using the `has_known_identity` trait class:
+```c++
+template <typename BinaryOperation, typename AccumulatorT>
+struct has_known_identity {
```

Comment:
Why do we need this? Isn't this the same as `is_detected<known_identity, BinaryOperation, AccumulatorT>`?

Reply:
@rarutyun argued that because `is_detected` isn't standard, it's nicer to expose a dedicated trait for this.

The reason this is different to the group algorithms fix is that non-expert users may still need to care about the identity values.  Reasoning about which form of the `reduction()` function to use and whether there may be some performance overhead is useful to application developers, not just library developers building an extra level of abstraction.

Reply:
I disagree. Not every problem needs to be solvable with just just SYCL and std. Users should use extra libraries for utilities.

Reply:
But it is a minor thing not worth arguing about. But I do think it would be better if we at least say that `has_known_identity<O,T>` is defined by whether `known_identity<O,T>::identity` exists. That means one can implement it using `is_detected` and in the future (when we add support for user to provide custom `known_identity`) users don't need to provide also `has_known_identity`.

Reply:
Good idea.  I've made the change in 0136df20a937.  I swapped the order in which the trait classes are introduced, so that `has_known_identity` can be defined in terms of `known_identity`, and tried to clarify that `known_identity::value` only exists if the value is known.